### ISSUE TITLE
nodejs nlp-with-luis msbot clone issues

### ISSUE THREAD

**Issue by JonathanFingold** [Microsoft/BotBuilder-Samples#1097]
## Sample information
1. Sample type: [samples]
2. Sample language: [nodejs]
3. Sample name: 12.nlp-with-luis

## Describe the bug
Following instructions for the msbot clone step produces a non-functional deployment.

## To Reproduce
1. Go to the [follow these steps](https://github.com/Microsoft/BotBuilder-Samples/tree/master/samples/javascript_nodejs/12.nlp-with-luis#to-try-this-sample) in the README.
2. Follow the steps.
3. Start the bot locally after the `msbot clone services` step completes successfully: `npm i` `npm start`.
4. See error:
```text
npm start

> nlp-with-luis@1.0.0 start ...\BotBuilder-Samples\samples\javascript_nodejs\12.nlp-with-luis
> node ./index.js

...\BotBuilder-Samples\samples\javascript_nodejs\12.nlp-with-luis\index.js:51
    endpoint: luisConfig.getEndpoint()
                         ^

TypeError: luisConfig.getEndpoint is not a function
    at Object.<anonymous> (...s\BotBuilder-Samples\samples\javascript_nodejs\12.nlp-with-luis\index.js:51:26)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
    at bootstrap_node.js:625:3
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nlp-with-luis@1.0.0 start: `node ./index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the nlp-with-luis@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     ...\AppData\Roaming\npm-cache\_logs\2018-12-18T18_04_07_782Z-debug.log
```

## Expected behavior
The bot should still work correctly.

## Additional context
- I was able to test the bot locally before deployment, and it worked fine.
- I deleted the original LUIS app before attempting the `msbot clone services` command.
- Starting the bot in debug mode in VSCode, I noticed that index.js line 45 was loading the wrong service:
  ```js
  const luisConfig = botConfig.findServiceByNameOrId(LUIS_CONFIGURATION);
  ```
  - This loads the "abs" service, which ends up with the same name via the msbot clone step.

[bug]

**Comment by JonathanFingold** [comment]
More context:
- my redacted msbot command:
```cmd
msbot clone services --name "jf1218-nlp-with-luis" --luisAuthoringKey xxxx --code-dir "." --location westus --sdkLanguage "Node" --folder deploymentScripts/msbotClone --verbose
```
- my resulting `jf1218-nlp-with-luis.bot` file. Note that the abs and luis service names are the same.
```json
{
    "name": "jf1218-nlp-with-luis",
    "description": "",
    "services": [
        {
            "type": "abs",
            "id": "3",
            "name": "jf1218-nlp-with-luis",
            "tenantId": "72f988bf-86f1-41af-91ab-2d7cd011db47",
            "subscriptionId": "0389857f-2464-451b-ac83-5f54d565b1a7",
            "resourceGroup": "jf1218-nlp-with-luis",
            "serviceName": "jf1218-nlp-with-luis",
            "appId": "518e5b86-1eb5-4538-9248-f3b931f5bae1"
        },
        {
            "type": "endpoint",
            "id": "1",
            "name": "development",
            "appId": "518e5b86-1eb5-4538-9248-f3b931f5bae1",
            "appPassword": "kzsk/dQxf+ayT6Qr6ANBbQ==!JguQm1Uog6f2WVytbxBRE4k9CgAcT6IrNMkjJ9SGeQ0=",
            "endpoint": "http://localhost:3978/api/messages"
        },
        {
            "type": "endpoint",
            "id": "2",
            "name": "production",
            "appId": "518e5b86-1eb5-4538-9248-f3b931f5bae1",
            "appPassword": "r3uNT34DD6jEDWMh9B+3RQ==!ahe5jQHWN/0SrZp+J1SboYl95mvjc3HBJVnGJ8OA7oU=",
            "endpoint": "https://jf1218nlpwithzomq.azurewebsites.net/api/messages"
        },
        {
            "type": "appInsights",
            "tenantId": "72f988bf-86f1-41af-91ab-2d7cd011db47",
            "subscriptionId": "0389857f-2464-451b-ac83-5f54d565b1a7",
            "resourceGroup": "jf1218-nlp-with-luis",
            "name": "jf1218-nlp-with-luisi44qrp",
            "serviceName": "jf1218-nlp-with-luisi44qrp",
            "instrumentationKey": "5z0sCX+mG3yaBzHim1WS7A==!1U7+mJT1cOB3R2cUXam6ig4SfmQz8+PC6u0ozXp9AnkXtMUX5S9ZU+13HQQyKuKu",
            "applicationId": "8996c548-6b36-4065-9319-6fbb5f35f8aa",
            "apiKeys": {},
            "id": "4"
        },
        {
            "type": "luis",
            "name": "jf1218-nlp-with-luis",
            "appId": "02c6192a-f4d0-42c6-b766-2a8c80cfaf55",
            "authoringKey": "J/+1H1t5Lk4IYtBgTllssg==!O8L+QRx+IlPVYGGUMwd+QUmmv2+tOHs8TTh4wrgEkCPDPTFAS1D9mhed8eimW8UB",
            "subscriptionKey": "mFIrftsi0MkGt4zQkJeOOg==!fLfLUkQ0nUAblThetFYC4WWnEj7IVDLczM+dRkxw0BYLRCKq1Qpv+YvpL9NRC1KQ",
            "version": "0.1",
            "region": "westus",
            "id": "34"
        },
        {
            "type": "blob",
            "name": "jf1218nlpl4zw",
            "serviceName": "jf1218nlpl4zw",
            "tenantId": "72f988bf-86f1-41af-91ab-2d7cd011db47",
            "subscriptionId": "0389857f-2464-451b-ac83-5f54d565b1a7",
            "resourceGroup": "jf1218-nlp-with-luis",
            "connectionString": "cRNzhltHX3VkxtPO8QXR4Q==!gtbUVKYE/aWHdn4K4PO2NYE+4zK31Jg0S0zy0szGjn1rReBBmq2qSkjS+daIJq4o2AMw5EcVWV4Wks+T8XqhbKWD0hZEXGnbsD9ucgCKmMTFJ8q1N2wphntjIaeJSn1nbPiVCLohBuRUWkKbxHEJgjXLB28MAJ0eJnwt8IMbZZrmtFbc3wN1PUmKkcsOZzDOBbE4P2FeZ0R8kFqaHqw8HjDQ9iCHgf7u9CfdKnCLMTtwH8IFLtoFV+xyuP7AdT7F",
            "container": null,
            "id": "190"
        }
    ],
    "padlock": "7aN0TiAIKJiF0q6VwpVqTQ==!T3gINU6xJP6kTP13kHhNz3KlAhl+2xCuYJ8ihc2SN1YfSGzjLMs47CcS4SYVZNID",
    "version": "2.0"
}
```

**Comment by sgellock** [comment]
Fixed with PR #1103
